02-intermediate-ruby/modules-as-namespaces.md

```diff
@@ -1,48 +1,115 @@
 # Modules as Namespaces
 
-### Namespacing
+## Learning Goals
+By the end of this lesson, students should be able to
 
-*Namespacing* is a way of bundling related objects together.  In ruby we use Modules to group related Classes and methods together.  We do this to avoid naming collisions.  
+- Read and write code using Ruby _modules_
+- Use modules to group classes and methods into _namespaces_
 
-For example we might create a class called "Shape" in a graphics program, but that name might be used elsewhere in another library.  When you do Shape.new, how is Ruby to know which Shape class to instantiate?  It's a similar idea to putting files into directories.  Files in the same directory cannot have the same name, but files in different directories may.  
+## Modules
 
-In this example below we can create an Array class and put it into a module.  Notice that it does not affect Ruby's default Array class at all
+Modules are a way to group together related ideas. They're like classes that can't be instantiated, making them a good choice if you need to tie together some methods that don't have any associated state.
+
+We've seen a module before: `Math`. It doesn't makes sense to have an instance of `Math`, it's just a collection of related methods and constants.
+
+```ruby
+# No instances required!
+Math::PI
+# => 3.141592653589793
+Math.sqrt(25)
+# => 5.0
+Math.cos(Math::PI / 4)
+# => 0.7071067811865476
+```
+
+### Syntax
+
+#### Defining a Module
+
+Defining a module is very similar to defining a class, except they use Ruby's `module` keyword.
```

Comment:
Might be good to add that any/all of these pieces of syntax are optional and not all required within a module.